x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -2358,4 +2358,9 @@ public void testFailingMetadataWithSquareBrackets() {
             "line 1:11: mismatched input '[' expecting {<EOF>, '|', ',', 'metadata'}"
         );
     }
+
+    public void testInvalidRemoteLookupJoin() {
+        // TODO ES-10559 this should be replaced with a proper error message once grammar allows indexPattern as joinTarget
+        expectError("FROM my-index | LOOKUP JOIN remote:languages_lookup ON language_code", "line 1:35: token recognition error at: ':'");
```

Comment:
I wanted to say that we should already start adding cases with quoting here, but realized that's kinda silly due to the grammar just not being there yet.

But maybe it makes sense to consider the associated issue blocked by ES-10559, just because we'll need to add more tests once the grammar is generalized. (In a follow-up PR)

I can think of
- `LOOKUP JOIN remote:"languages_lookup"`
- `LOOKUP JOIN "remote:languages_lookup"`
- `LOOKUP JOIN """remote:languages_lookup"""`
- `LOOKUP JOIN remote:"""languages_lookup"""`
and I'm unsure if all of these actually make sense.

Reply:
I am likely going to merge this after https://github.com/elastic/elasticsearch/pull/120494 (that is changing the same classes). I will add above variations once resolving the conflicts.

Reply:
Since #120494 adds nice options for randomization, we could also use randomization here.

Other test cases we could consider:
- multiple colons
- colons and wildcards
- colons and date math - especially because we want to enable date math later, and valid date math expressions themselves can contain colons
- quoting (`"` and `"""`) of the remote name + 1 or multiple colons - just so we can later see how the error message changes from a token recognition error to a proper parsing exception once https://github.com/elastic/elasticsearch/pull/120355 lands.